src/SprykerEco/Yves/Braintree/Theme/default/components/molecules/payment-form/payment-form.ts

```diff
@@ -0,0 +1,150 @@
+declare var braintree: any;
+
+import Component from 'ShopUi/models/component';
+import ScriptLoader from 'ShopUi/components/molecules/script-loader/script-loader';
+
+interface braintreeConfig {
+    nonce: string,
+    type: string
+}
+
+interface braintreeErrorConfig {
+    message: string
+}
+
+export default class PaymentForm extends Component {
+    form: HTMLFormElement;
+    paymentMethods: HTMLInputElement[];
+    nonceInputSelector: HTMLInputElement;
+    braintreeSetupSettings: any;
+    scriptLoader: ScriptLoader;
+    currentPaymentMethodValue: string = '';
+
+    readonly formId: string = 'payment-form';
+    readonly nonceInputName: string = 'payment_method_nonce';
+    readonly paymentSelection: string = 'paymentForm[paymentSelection]';
+    readonly integrationType: string = 'custom';
+    readonly paymentMethodName: string = '';
+    readonly paymentMethodTypeName: string = '';
+
+    constructor() {
+        super();
+    }
+
+    protected readyCallback(): void {
+        this.form = <HTMLFormElement>document.getElementById(`${this.formId}`);
+        this.paymentMethods = <HTMLInputElement[]>Array.from(this.form.querySelectorAll(`input[name='${this.paymentSelection}']`));
+        this.nonceInputSelector = <HTMLInputElement>this.querySelector(`input[name='${this.nonceInputName}']`);
+        this.scriptLoader = <ScriptLoader>this.querySelector('script-loader');
+
+        this.setCurrentPaymentMethod();
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.scriptLoader.addEventListener('scriptload', () => this.onScriptLoad());
+
+        this.paymentMethods.forEach((method: HTMLInputElement) => {
+            method.addEventListener('change', () => {
+                this.clearNonceInputSelector();
+                this.setCurrentPaymentMethodValue(method);
+            });
+        });
+    }
+
+    protected onScriptLoad(): void {
+        if (!!this.braintreeClientToken) {
+            this.setupBraintree();
+        }
+    }
+
+    protected submitForm(nonce: string = ''): void {
+        this.nonceInputSelector.value = nonce;
+
+        this.form.submit();
+    }
+
+    protected errorHandler(error: braintreeErrorConfig): any {
+        const paymentMethod = this.currentPaymentMethodValue;
+
+        this.removeErrorMessage();
+
+        if (paymentMethod === this.paymentMethodName) {
+            return this.addErrorMessage(error.message);
```

Comment:
Strange return, why do you need to return something?

Reply:
it prevent form submit if there is an error

Reply:
removed